### ISSUE TITLE
Expert Weight Collapse in Evolutionary Merging (TIES/CMA-ES) with OpenBioLLM

### ISSUE THREAD

**Issue by Zheng-Jay** [hammoudhasan/MergeAlign#1]
Hello, and thank you for your outstanding contributions to MergeAlign! I’m attempting to reproduce the model evolution experiments (TIES-Merging and CMA-ES) described in the paper, but I’m observing some unexpected behavior: the optimizer consistently drives the expert model’s contribution toward zero.

Experiment setup:

Task: Reproducing the experiments involving {LLaMA, OpenBioLLM}.

Base model: Meta-Llama-3-8B-Instruct

Expert model: Llama3-OpenBioLLM-8B

Methods: TIES-Merging and CMA-ES

Training data: Safety data and expert data (generated by the scripts provided in the repository, using the safety model and expert model respectively)

Observed issue:
In my experiments, the evolutionary algorithm pushes the expert model’s weight toward negative values or zero, indicating that even on data generated by the expert model itself, the expert model shows higher perplexity (loss) than the base model.

Loss setup:

Full loss: expert loss + safety loss

Expert loss only

I tested two configurations for the expert weight boundaries:

Unbounded: For both loss types, the expert weight converges to around –1.4.

Positive boundary ([0, 1.0]): For both loss types, the expert weight converges to ~0.0 (effectively discarding the expert).

Additional question:
I also noticed that when evaluating OpenBioLLM using the beaver_tails safety evaluation set mentioned in the paper, OpenBioLLM tends to repeat the question (even when adding the chat template). How did you handle this behavior in your evaluations?

<img width="1639" height="331" alt="Image" src="https://github.com/user-attachments/assets/8f7820f0-5ab2-4c80-867c-34a44ae9f999" />
